### ISSUE TITLE
Question about the learnability defination.

### ISSUE THREAD

**Issue by Yxxxb** [microsoft/DELT#12]
Thank you for your excellent work!

<img width="1281" height="112" alt="Image" src="https://github.com/user-attachments/assets/d0ed909f-0930-435c-a344-39cd5e691071" />

I'm curious about how to calculate learnability. Formula 6 defines the learnability of a sample xn. I find that it is cumulative over training steps t from 1 to T. However, pre-training typically uses only one epoch, which means that each sample is trained only once. So how can I obtain the gradient change of xn for each of the T steps?

Thanks.

**Comment by Yxxxb** [comment]
@dialogueeeeee @huangyangyu 
Thanks in advance for your answers.

**Comment by dialogueeeeee** [comment]
Thank you for your interest in our work. @Yxxxb 

The process of scoring samples (calculating learnability) is separate from the model training process (pre-training for one epoch). We first score all the samples, and then perform selection, ordering, and model training separately, rather than dynamically scoring samples during model training.

For a detailed explanation of the scoring process, please refer to **Appendix E. LQS Implementation** in our paper.
The corresponding code implementation for LQS calculation can be found in `data_scoring/lqs/annotation/` .
